### PR TITLE
Promoted (most of) the supported cloud identity resources to GA

### PR DESCRIPTION
**Pull request by melinath**
Resolved https://github.com/hashicorp/terraform-provider-google/issues/7757.

Notes:

- Enabled additional_keys and member_key fields, beta-only
- Removed references to dynamic groups; they are in the API but will require additional work because they require a premium SKU to test. Since they were already not included in the provider, we don't lose anything by fully deleting them.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloud_identity: promoted `google_cloud_identity_group` to GA
```
```release-note:enhancement
cloud_identity: promoted `google_cloud_identity_group_membership` to GA
```
```release-note:enhancement
cloud_identity: promoted data source `google_cloud_identity_groups` to GA
```
```release-note:enhancement
cloud_identity: promoted data source `google_cloud_identity_group_memberships` to GA
```

### CODE REVIEW

**Comment by melinath** [comment]
non-beta test run here: https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=156843

File: templates/terraform/examples/cloud_identity_group_membership_user.tf.erb [hunk 1]
```diff
@@ -14,10 +13,9 @@ resource "google_cloud_identity_group" "group" {
 }
 
 resource "google_cloud_identity_group_membership" "cloud_identity_group_membership_basic" {
-  provider = google-beta
   group    = google_cloud_identity_group.group.id
 
-  member_key {
+  preferred_member_key {
```
Comment (slevenick) [inline templates/terraform/examples/cloud_identity_group_membership_user.tf.erb:18]: Why is the name changing here?

Comment (melinath) [reply templates/terraform/examples/cloud_identity_group_membership_user.tf.erb:18]: member_key is only supported in the beta API. This test is being run against the GA API; member_key usage is instead handled by handwritten tests in third_party/terraform/tests/resource_cloud_identity_group_membership_test.go.erb

File: products/cloudidentity/api.yaml [hunk 2]
```diff
@@ -15,6 +15,9 @@
 name: CloudIdentity
 display_name: Cloud Identity
 versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://cloudidentity.googleapis.com/v1beta1/
   - !ruby/object:Api::Product::Version
```
Comment (slevenick) [inline products/cloudidentity/api.yaml:21]: This seems wrong

File: products/cloudidentity/api.yaml [hunk 3]
```diff
@@ -222,6 +143,7 @@ objects:
         input: true
         description: |
           EntityKey of the member.
+        min_version: beta
```
Comment (slevenick) [inline products/cloudidentity/api.yaml:146]: How does `exactly_one_of` work when one field is beta and one isn't?

**Review by slevenick — APPROVED** [review]

File: products/cloudidentity/api.yaml (hunk 2 above)
Comment (melinath) [reply products/cloudidentity/api.yaml:21]: what seems wrong about it?

Comment (melinath) [reply products/cloudidentity/api.yaml:21]: ah I put in the wrong url thanks for catching!

Comment (melinath) [reply products/cloudidentity/api.yaml]: Fixed in 69e31119bce49986d1671ed0807781ac98ab7d66

File: products/cloudidentity/api.yaml (hunk 3 above)
Comment (melinath) [reply products/cloudidentity/api.yaml:146]: It looks like [exactly_one_of is converted directly into terraform](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/templates/terraform/schema_property.erb#L178), but using [get_property_schema_path](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/provider/terraform.rb#L129) and `array.compact`, which seems to be [eliminating the beta field from the resulting golang code](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-4211-old..auto-pr-4211#diff-8774cce34c1912529ae270958b564cb43298e2afc52e4b7e44b578d96a3b70faR106).

In other words, in GA, the go code ends up with a "exactly one of" constraint that doesn't include beta fields - in this case meaning it only contains one property, at least one of which much be specified, which I believe is the correct behavior.

**Comment by melinath** [comment]
Everything still passes after correcting the GA url so I'm gonna go ahead and merge.

https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=157312
https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=157314

**Review by rileykarson — COMMENTED** [review]
I think we need to un-guard these lines as well, to make the datasources appear at GA: https://github.com/GoogleCloudPlatform/magic-modules/blob/master/third_party/terraform/utils/provider.go.erb#L183-L186

That should additionally resolve the unused code warnings in tpg eg https://travis-ci.org/github/hashicorp/terraform-provider-google/builds/743049795

**Comment by melinath** [comment]
Thanks Riley - I've opened #4229 making that change!
